src/storage-manager.coffee

```diff
@@ -205,8 +203,19 @@ class _StorageManager
   #
   # Private
 
-  _invalidateCache: (name) ->
-    @collections[name].cache = {}
+  _shouldInvalidateCache: (model) ->
+    return false if !model
+    return false if model.isNew()
+    return true if Object.keys(model.changed).length is 0 # destroyed
```

Comment:
This can check `model.hasChanged()` since it's logically equivalent.

Reply:
Good call!